app/models/driver.rb

```diff
@@ -0,0 +1,38 @@
+class Driver < ApplicationRecord
+  has_many :trips, dependent: :nullify
+
+  validates :name, presence: true
+  validates :vin, uniqueness: true
+
+  def rating
+    num_trips = self.trips.length
+    ongoing_trips = 0
+    rating = 0
+
+    num_trips.times do |i|
+      if self.trips[i].rating != nil
+        rating += self.trips[i].rating
+      else
+        ongoing_trips += 1
+      end
+    end
+
+    rating = rating.to_f/(num_trips-ongoing_trips)
```

Comment:
you end up needing the `ongoing_trips` variable to keep track of the length of this collection... Maybe there's a different way to filter or select ( https://ruby-doc.org/core-2.5.1/Enumerable.html#method-i-select ), and use that collection...